util/parsenum.h

```diff
@@ -121,8 +121,10 @@ _Pragma("clang diagnostic pop")
 			    ((min) <= 0) ? 0 : (min),			\
 			    (uintmax_t)(max), (*(x)), (base),		\
 			    (trailing))),				\
-				((((max) < 0) && (errno == 0)) ?	\
-				    (errno = ERANGE) : 0)),		\
+				(((max) <= INTMAX_MAX) ?		\
```

Comment:
Not sure what you mean by "if the compiler has to treat (max) as an unsigned value" here.  Does this help if max is an unsigned variable?

Reply:
Right, it took me a while to decypher the commit message.  The idea is basically sanity-checking the upcoming cast to `intmax_t`.  I mean, if max is `INTMAX_MAX + 1`, then we can't do `((intmax_t)(max) < 0) ?`

... at the time, I thought that I had a reason to split up the change into two commits, but at the moment I'm not recreating my thought process.